require.js

```diff
@@ -49,10 +49,29 @@
     }
 
     var globalEval = eval; // reassigning causes eval to not use lexical scope.
+    var ArrayPush = Array.prototype.push;
 
     // Non-CommonJS speced extensions should be marked with an "// EXTENSION"
     // comment.
 
+
+	var _Module = function _Module() {};
+	_Module.prototype.id = null;
+	_Module.prototype.display = null;
+	_Module.prototype.require = null;
+	_Module.prototype.factory = void 0;
+	_Module.prototype.exports = void 0;
+	_Module.prototype.redirect = void 0;
+	_Module.prototype.location = null;
+	_Module.prototype.directory = null;
+	_Module.prototype.injected = false;
+	_Module.prototype.mappingRedirect = void 0;
+	_Module.prototype.type = null;
+	_Module.prototype.text = void 0;
+	_Module.prototype.dependees = null;
+	_Module.prototype.extraDependencies = void 0;
+	_Module.prototype.uuid = null;
```

Comment:
That's just V8 though!


montagestudio.com
 	
Benoit Marchant
CEO & Founder
1290 Oakmead Parkway,y Ste 111
Sunnyvale, CA 94085
(1) 408.621.4874 mobile
Profile | @benoitmarchant

> On Oct 6, 2015, at 19:14, Kris Kowal <notifications@github.com> wrote:
> 
> In require.js:
> 
> > +	var _Module = function _Module() {};
> > +	_Module.prototype.id = null;
> > +	_Module.prototype.display = null;
> > +	_Module.prototype.require = null;
> > +	_Module.prototype.factory = void 0;
> > +	_Module.prototype.exports = void 0;
> > +	_Module.prototype.redirect = void 0;
> > +	_Module.prototype.location = null;
> > +	_Module.prototype.directory = null;
> > +	_Module.prototype.injected = false;
> > +	_Module.prototype.mappingRedirect = void 0;
> > +	_Module.prototype.type = null;
> > +	_Module.prototype.text = void 0;
> > +	_Module.prototype.dependees = null;
> > +	_Module.prototype.extraDependencies = void 0;
> > +	_Module.prototype.uuid = null;
> null properties on the prototype do not improve performance or self-documentation. Assigning properties in the constructor does guarantee a v8 hidden class and serves to document the intended properties of the instances.
> 
> —
> Reply to this email directly or view it on GitHub.
>